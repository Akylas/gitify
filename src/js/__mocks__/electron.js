window.Notification = function (title, options) {
  this.title = title;

  return {
    onclick: jest.fn()
  };
};

window.Audio = class Audio {
  constructor(path) {
    this.path = path;
  }

  play() {}
};

window.localStorage = {
  store: {},
  getItem: function (key) { return this.store[key]; },
  setItem: function (key, item) {
    this.store[key] = item;
  },
  removeItem: jest.fn()
};

window.alert = jest.fn();

const browserWindow = {
  loadURL: jest.fn(),
  webContents: {
    on: () => {},
  },
  on: () => {},
  close: jest.fn(),
  destroy: jest.fn()
};

module.exports = {
  remote: {
    BrowserWindow: () => browserWindow
  },
  ipcRenderer: {
    send: jest.fn()
  },
  shell: {
    openExternal: jest.fn()
  },
};
