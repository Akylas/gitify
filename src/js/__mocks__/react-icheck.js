import React from 'react';

export class Checkbox extends React.PureComponent {
  render() {
    return (
      <div>
        <strong>{this.props.label}</strong>
        <input
          type="checkbox"
          checked={this.props.defaultChecked}
          onChange={(evt) => this.props.onChange(evt)} />
      </div>
    );
  };
}

export class Radio extends React.PureComponent {
  render() {
    return (
      <div>
        <strong>{this.props.label}</strong>
        <input
          type="radio"
          value={this.props.value}
          checked={this.props.defaultChecked}
          onChange={(evt) => this.props.onChange(evt)} />
      </div>
    );
  };
}

export class RadioGroup extends React.PureComponent {
  render() {
    return (
      <div>
        {this.props.children}
      </div>
    );
  };
}
