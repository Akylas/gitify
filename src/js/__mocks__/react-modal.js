import React from 'react';

export default class Modal extends React.PureComponent {
  render() {
    return this.props.isOpen ? (
      <div>{this.props.children}</div>
    ) : null;
  };
}
