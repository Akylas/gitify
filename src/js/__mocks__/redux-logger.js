module.exports = function createLogger () {
  return ({ getState }) => (next) => (action) => {
    return next(action);
  };
};
