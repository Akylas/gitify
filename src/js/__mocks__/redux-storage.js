import { Map } from 'immutable';

export function createMiddleware(engine, actionBlacklist = [], actionWhitelist = []) {
  return ({ dispatch, getState }) => {
    return (next) => (action) => {
      return next(action);
    };
  };
};

export function createLoader() {
  const state = {
    auth: Map({
      token: 'LOGGED_IN_TOKEN'
    })
  };

  return () => new Promise(resolve => resolve(state));
}

export function reducer (clb) {
  return (state, action) => clb(state, action);
};
