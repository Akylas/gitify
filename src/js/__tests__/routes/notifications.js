import React from 'react'; // eslint-disable-line no-unused-vars
import { Map } from 'immutable';
import renderer from 'react-test-renderer';

import { NotificationsRoute, mapStateToProps } from '../../Routes/Notifications';
import { mockedNotificationsRecuderData } from '../../__mocks__/mockedData';

jest.mock('../../components/notifications');

describe('routes/notification.js', () => {
  const props = {
    failed: false,
    accountNotifications: mockedNotificationsRecuderData
  };

  it('should test the mapStateToProps method', () => {
    const state = {
      notifications: Map({
        response: mockedNotificationsRecuderData,
        failed: false,
      }),
    };

    const mappedProps = mapStateToProps(state);

    expect(mappedProps.failed).toBeFalsy();
    expect(mappedProps.accountNotifications).toEqual(mockedNotificationsRecuderData);
  });

  it('should render itself & its children', () => {
    const tree = renderer.create(
      <NotificationsRoute {...props} />
    );

    expect(tree).toMatchSnapshot();
  });
});
