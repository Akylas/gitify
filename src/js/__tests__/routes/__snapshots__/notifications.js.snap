// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/notification.js should render itself & its children 1`] = `
<div
  className="container-fluid main-container notifications"
>
  <span />
</div>
`;
